src/openssl/ssl/server.cr

```diff
@@ -0,0 +1,42 @@
+require "socket"
+
+class OpenSSL::SSL::Server
+  include ::Socket::Server
+
+  getter wrapped : ::Socket::Server
+  getter context : OpenSSL::SSL::Context::Server
+
+  # If `#sync_close?` is `true`, closing this server will
+  # close the wrapped server.
+  property? sync_close : Bool
+
+  getter? closed : Bool = false
+
+  def initialize(@wrapped : ::Socket::Server, @context : OpenSSL::SSL::Context::Server = OpenSSL::SSL::Context::Server.new, @sync_close : Bool = true)
+  end
+
+  def self.open(wrapped : ::Socket::Server, context : OpenSSL::SSL::Context::Server = OpenSSL::SSL::Context::Server.new, sync_close : Bool = true)
+    server = new(wrapped, context, sync_close)
+
+    begin
+      yield server
+    ensure
+      server.close
+    end
+  end
+
+  def accept? : OpenSSL::SSL::Socket::Server?
+    if socket = @wrapped.accept?
+      OpenSSL::SSL::Socket::Server.new(socket, @context, sync_close: @sync_close)
+    end
+  end
+
+  def close
+    return if @closed
+    @closed = true
+
+    @wrapped.close if @sync_close
+  end
+
+  delegate local_address, remote_address, to: @wrapped
```

Comment:
If you're going to add these delegations you might as well just make `wrapped` a `Socket` instead of a `Socket::Server`.

Reply:
Actually, `remote_address` is incorrect, a server doesn't have one. Only sockets do.
But every server should have a `remote_address` (see my proposal in #5778). So I think `@wrapped`'s type should stay `Socket::Server`. And this way you can wrap SSL socket servers ;)